src/scripts/stars.js

```diff
@@ -0,0 +1,63 @@
+const section = document.querySelector(".section--products");
+const starBoxes = document.querySelectorAll(".stars");
+
+section.addEventListener("mouseover", function(e) {
+    for (i=0; i<=4; i++) {
+        if (event.target.classList.contains(`s${i+1}`)) {
+            selectStars(i);
+        }
+        else if (event.target.classList.contains('stars')) {
+            deselectStars();
+        }   
+    }
+});
```

Comment:
Ta petla for to wciąż nie jest najwydajniejsze rozwiązanie. Nie potrzebnie sprawdzamy wszystkie elementy. Do tego robimy to nawet jesli pasował nam warunek już dla i=0.  Wydaje mi się, że można by ten cały skrypt zdecydowanie skrócić. Zastanów się nad najprostszym rozwiązaniem :)